content/storage-history/history.textile

```diff
@@ -0,0 +1,596 @@
+    if events_page.has_next?
+    next_page = events_page.next
+    puts "#{next_page.items.length} presence events received on second page"
+    end
+
+    bc[php]. $eventsPage = $channel->presence->history();
+    echo(count($eventsPage->items) . ' presence events received in first page');
+    if ($eventsPage.hasNext()) {
+    $nextPage = $eventsPage->next();
+    echo(count($nextPage->items) . ' presence events received in second page');
+    }
+
+    ```[python]
+    events_page = channel.presence.history()
+    print str(len(events_page.items)) + " presence events received"
+    if events_page.has_next():
+    next_page = events_page.next()
+    ```
+
+    ```[java]
+    PaginatedResult<PresenceMessage> eventsPage = channel.presence.history(null);
+    System.out.println(eventsPage.items().length + " presence events received in first page");
+    if(eventsPage.hasNext()) {
+    PaginatedResult<PresenceMessage> nextPage = eventsPage.next();
+    System.out.println(nextPage.items().length + " presence events received in 2nd page");
+    }
+    ```
+
+    ```[csharp]
+    PaginatedResult<PresenceMessage> eventsPage = await channel.Presence.HistoryAsync();
+    Console.WriteLine(eventsPage.Items.Count + " presence events received in first page");
+    if (eventsPage.HasNext)
+    {
+        PaginatedResult<PresenceMessage> nextPage = await eventsPage.NextAsync();
+        Console.WriteLine(nextPage.Items.Count + " presence events received in 2nd page");
+    }
+    ```
+
+    ```[objc]
+    [channel.presence history:^(ARTPaginatedResult<ARTPresenceMessage *> *eventsPage, ARTErrorInfo *error) {
+    NSLog(@"%lu presence events received in first page", [eventsPage.items count]);
+    if (eventsPage.hasNext) {
+        [eventsPage next:^(ARTPaginatedResult<ARTPresenceMessage *> *nextPage, ARTErrorInfo *error) {
+        NSLog(@"%lu presence events received in 2nd page", [nextPage.items count]);
+        }];
+    }
+    }];
+    ```
+
+    ```[swift]
+    channel.presence.history { eventsPage, error in
+    let eventsPage = eventsPage!
+    print("\(eventsPage.items.count) presence events received in first page")
+    if eventsPage.hasNext {
+        eventsPage.next { nextPage, error in
+        print("\(nextPage!.items.count) presence events received in 2nd page")
+        }
+    }
+    }
+    ```
+
+    ```[go]
+    page, err := channel.Presence.History(nil)
+    fmt.Println("%d messages in first page\n", len(page.PresenceMessages()))
+    if page.hasNext() {
+    page2, err := page.Next()
+    fmt.Println("%d messages on 2nd page!\n", len(page2.PresenceMessages()))
+    }
+    ```
+
+This gets a paginated set of historical presence messages for the channel. If the channel is configured to persist messages to disk, then message history will typically be available for 24 or 72 hours. If not, messages are only retained in memory by the Ably service for two minutes. See "persistence enabled":/storage-history/storage.
```

Comment:
This seems repetitive and inconsistent with the info offered for channel history.